### PR TITLE
thuang-update-smoke-test

### PR DESCRIPTION
**Pull request by tihuan**
Updated the files 😄 👍 Thanks for the great suggestion, @Bento007 ! 🙏 🚀

### CODE REVIEW

File: frontend/Makefile [hunk 1]
```diff
@@ -22,10 +24,19 @@ clean:
...
+	$(check-deployment-stage-flag)
 	aws s3 cp $(S3_ENVIRONMENT_FILE) .env.production
 	# Copy the appropriate configs
 	cp src/configs/$(DEPLOYMENT_STAGE).js src/configs/configs.js
 
 .PHONY: upload-vars
 upload-vars:
 	aws s3 cp .env.production $(S3_ENVIRONMENT_FILE)
+
+.PHONY: smoke-test-with-local-backend
```
Comment (tihuan) [inline frontend/Makefile:36]: export two `npm commands` here for the top level `Make` file to use

File: frontend/Makefile [hunk 2]
```diff
@@ -3,9 +3,11 @@ ACCOUNT_ID=$(shell aws sts get-caller-identity --query Account --output text)
 BROWSER_S3=corpora-static-site-$(DEPLOYMENT_STAGE)-$(ACCOUNT_ID)
 S3_ENVIRONMENT_FILE=s3://$(BROWSER_S3)/env.production
 
-ifndef DEPLOYMENT_STAGE
-$(error Please set the DEPLOYMENT_STAGE environment before deploying)
-endif
+define check-deployment-stage-flag
```
Comment (tihuan) [inline frontend/Makefile:6]: wrap this into a function, since only `retrieve-vars` uses it

File: .github/workflows/push-tests.yml [hunk 3]
```diff
@@ -82,9 +82,8 @@ jobs:
           cp src/configs/local.js src/configs/configs.js
           npm install
       - name: smoke test prod build with local BE server
-        working-directory: ./frontend
         run: |
-          npm run smoke-test-with-local-backend-ci
+          make smoke-test-with-local-backend-ci
```
Comment (tihuan) [inline .github/workflows/push-tests.yml:86]: use `Make` file's command instead for consistency!

File: Makefile [hunk 4]
```diff
@@ -35,3 +35,11 @@ local-database: clean_test_db
 .PHONY: local-backend
 local-backend:
 	$(MAKE) local-server -C ./backend/chalice/api_server
+
+.PHONY: smoke-test-with-local-backend
+smoke-test-with-local-backend:
+	$(MAKE) smoke-test-with-local-backend -C ./frontend
+
+.PHONY: smoke-test-with-local-backend-ci
+smoke-test-with-local-backend-ci:
```
Comment (tihuan) [inline Makefile:44]: this is used in Github Action

File: Makefile [hunk 5]
```diff
@@ -35,3 +35,11 @@ local-database: clean_test_db
 .PHONY: local-backend
 local-backend:
 	$(MAKE) local-server -C ./backend/chalice/api_server
+
+.PHONY: smoke-test-with-local-backend
+smoke-test-with-local-backend:
```
Comment (tihuan) [inline Makefile:40]: This allows people to run `make smoke-test-with-local-backend` in the root directory

**Review by Bento007 — APPROVED** [review]
LGTM

**Pull request merged**